workbench/src/dialogs.c

```diff
@@ -0,0 +1,411 @@
+	gtk_file_filter_add_pattern (filter, "*");
+	gtk_file_chooser_add_filter (GTK_FILE_CHOOSER(dialog), filter);
+
+	if (gtk_dialog_run(GTK_DIALOG(dialog)) == GTK_RESPONSE_ACCEPT)
+	{
+		gchar *locale_filename = gtk_file_chooser_get_filename(GTK_FILE_CHOOSER(dialog));
+		utf8_filename = utils_get_utf8_from_locale(locale_filename);
+
+		g_free(locale_filename);
+	}
+
+	gtk_widget_destroy(dialog);
+
+	return utf8_filename;
+}
+
+
+/** Shows the dialog "Add directory".
+ *
+ * The dialog lets the user choose an existing folder.
+ *
+ * @return The filename
+ *
+ **/
+gchar *dialogs_add_directory(WB_PROJECT *project)
+{
+	gchar *utf8_filename = NULL;
+	GtkWidget *dialog;
+
+	dialog = gtk_file_chooser_dialog_new(_("Add directory"),
+		GTK_WINDOW(wb_globals.geany_plugin->geany_data->main_widgets->window), GTK_FILE_CHOOSER_ACTION_SELECT_FOLDER,
+		_("_Cancel"), GTK_RESPONSE_CANCEL,
+		_("Add"), GTK_RESPONSE_ACCEPT, NULL);
+	if (project != NULL)
+	{
+		gchar *path;
+
+		/* Set the current folder to the location of the project file */
+		path = wb_project_get_filename(project);
+		if (path != NULL)
+		{
+			path = g_path_get_dirname(path);
+			gtk_file_chooser_set_current_folder(GTK_FILE_CHOOSER(dialog), path);
+			g_free(path);
+		}
+	}
+
+	if (gtk_dialog_run(GTK_DIALOG(dialog)) == GTK_RESPONSE_ACCEPT)
+	{
+		gchar *locale_filename = gtk_file_chooser_get_filename(GTK_FILE_CHOOSER(dialog));
+		utf8_filename = utils_get_utf8_from_locale(locale_filename);
+
+		g_free(locale_filename);
+	}
+
+	gtk_widget_destroy(dialog);
+
+	return utf8_filename;
+}
+
+
+/* Split patterns in str into gchar** */
+static gchar **split_patterns(const gchar *str)
+{
+	GString *tmp;
+	gchar **ret;
+	gchar *input;
+
+	input = g_strdup(str);
+
+	g_strstrip(input);
+	tmp = g_string_new(input);
+	g_free(input);
+	do {} while (utils_string_replace_all(tmp, "  ", " "));
+	ret = g_strsplit(tmp->str, " ", -1);
+	g_string_free(tmp, TRUE);
+	return ret;
+}
+
+
+/** Shows the directory settings dialog.
+ *
+ * The dialog lets the user edit the settings for file patterns, irnored file patterns and
+ * ignored directories patterns. On accept the result is directly stored in @a directory.
+ *
+ * @param  directory Location of WB_PROJECT_DIR to store the settings into
+ * @return TRUE if the settings have changed, FALSE otherwise
+ *
+ **/
+gboolean dialogs_directory_settings(WB_PROJECT_DIR *directory)
+{
+	GtkWidget *w_file_patterns, *w_ignored_dirs_patterns, *w_ignored_file_patterns;
+	GtkWidget *dialog, *label, *content_area;
+	GtkWidget *vbox, *hbox, *hbox1, *table;
+	GtkDialogFlags flags;
+	gchar *file_patterns_old, *ignored_file_patterns_old, *ignored_dirs_patterns_old;
+	gboolean changed;
+
+	/* Create the widgets */
+	flags = GTK_DIALOG_MODAL | GTK_DIALOG_DESTROY_WITH_PARENT;
+	dialog = gtk_dialog_new_with_buttons (_("Directory settings"),
+										  GTK_WINDOW(wb_globals.geany_plugin->geany_data->main_widgets->window),
+										  flags,
+										  GTK_STOCK_CANCEL,
```

Comment:
Maybe I have found it in some other code and then started to use it without changing the previous code. But I removed it because it is deprecated. There doesn't seem to be a replacement.